crates/api_common/src/request.rs

```diff
@@ -41,12 +44,45 @@ pub fn client_builder(settings: &Settings) -> ClientBuilder {
     .user_agent(user_agent.clone())
     .timeout(REQWEST_TIMEOUT)
     .connect_timeout(REQWEST_TIMEOUT)
+    .redirect(Policy::none())
```

Comment:
Instead of disabling redirects here and checking the url before using the client, move the check into a function that returns err to prevent the request (like `asysc fn check_url(request: Request) -> LemmyResult<Request>`), and use it here like this:

```suggestion
    .connector_layer(tower::filter::AsyncFilterLayer::new(check_url))
```